employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/pages/rotation/RotationPageViewportBuilder.java

```diff
@@ -143,30 +146,30 @@ private long getCurrentWorkerStartTime() {
+
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                            shiftTemplateViewList.forEach(stv -> spotIdToShiftTemplateViewListMap.get(stv.getSpotId()).add(stv));
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                            rotationView.setSpotIdToShiftTemplateViewListMap(spotIdToShiftTemplateViewListMap);
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                            res.onInvoke(rotationView);
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                        }
+                                                                                                                                                                                                                                                                                                                                                                                                                                          ))
+                                                                                                                                                                                                                                                                       ))
+                                                                                                                                                            ))
+                                            ))
```

Comment:
Oops, we have a formatting problem here. I checked the code out and verified that this is actually formatted "correctly" so we need to be creative to reduce the line length. This is my proposal:
```java

        return promiseUtils.promise((res, rej) -> RosterRestServiceBuilder.getRosterState(
                tenantStore.getCurrentTenantId(),
                restCallbackFactory.onSuccess(rosterState -> SpotRestServiceBuilder.getSpotList(
                        tenantStore.getCurrentTenantId(),
                        restCallbackFactory.onSuccess(spotList -> EmployeeRestServiceBuilder.getEmployeeList(
                                tenantStore.getCurrentTenantId(),
                                restCallbackFactory.onSuccess(employeeList -> RotationRestServiceBuilder.getShiftTemplateList(
                                        tenantStore.getCurrentTenantId(),
                                        restCallbackFactory.onSuccess(shiftTemplateViewList -> {
                                                                          RotationView rotationView = new RotationView();
                                                                          rotationView.setTenantId(tenantStore.getCurrentTenantId());
                                                                          rotationView.setRotationLength(rosterState.getRotationLength());
                                                                          rotationView.setSpotList(spotList);
                                                                          rotationView.setEmployeeList(employeeList);
                                                                          Map<Long, List<ShiftTemplateView>> spotIdToShiftTemplateViewListMap = new HashMap<>();
                                                                          spotList.forEach(spot -> spotIdToShiftTemplateViewListMap.put(spot.getId(), new ArrayList<>()));

                                                                          shiftTemplateViewList.forEach(stv -> spotIdToShiftTemplateViewListMap.get(stv.getSpotId()).add(stv));
                                                                          rotationView.setSpotIdToShiftTemplateViewListMap(spotIdToShiftTemplateViewListMap);
                                                                          res.onInvoke(rotationView);
                                                                      }
                                        ))
                                ))
                        ))
                ))
        );
```

Reply:
That looks pretty similar to what it was before I used the Formatter. Hopefully it doesn't change back again.

Reply:
The main difference is that the two parameters (id+callback) of each service call are always on a new line. That reduces the indentation accumulation. When I hit autoformat in my IDEA it stays unchanged. I try to follow KIE conventions as close as possible so I hope it will work for you as well.

Reply:
Fixed; thanks!